backend/curriculum_tracking/tests/test_activity_log_entry_creators_envoked.py

```diff
@@ -59,8 +67,162 @@ def test_start_project(self):
 # class log_card_review_request_cancelled_Tests(TestCase):
 
 
-# class log_card_moved_to_complete_Tests(TestCase):
+class log_card_moved_to_complete_Tests(TestCase):
+
+    def test_card_to_complete_creates_one_log_entry(self):
+
+        actor_user = UserFactory(is_superuser=True)
+        card = factories.AgileCardFactory(
+            status=AgileCard.COMPLETE,
+            content_item=factories.ProjectContentItemFactory(
+                project_submission_type=ContentItem.LINK, template_repo=None
+            ),
+        )
+        card.save()
+
+        creators.log_card_moved_to_complete(card, actor_user)
+        self.assertTrue(card.status==AgileCard.COMPLETE)
+        self.assertEqual(LogEntry.objects.count(), 1)
+        self.assertEqual(LogEntry.objects.first().actor_user, actor_user)
+
+    def test_card_to_complete_then_rf_then_complete_creates_one_complete_entry_within_debounce_period(self):
+        """
+        Although two LogEntries are created, only one of them is of status 'CARD_MOVED_TO_COMPLETE'
+        """
+        actor_user = UserFactory(is_superuser=True)
+        card = factories.AgileCardFactory(
+            status=AgileCard.COMPLETE,
+            content_item=factories.ProjectContentItemFactory(
+                project_submission_type=ContentItem.LINK, template_repo=None
+            ),
+        )
+
+        card.save()
+        creators.log_card_moved_to_complete(card, actor_user)
+        self.assertTrue(card.status == AgileCard.COMPLETE)
+        self.assertEqual(LogEntry.objects.count(), 1)
+
+        card.status = AgileCard.REVIEW_FEEDBACK
+        card.save()
+        creators.log_card_moved_to_review_feedback(card, actor_user)
+        self.assertTrue(card.status == AgileCard.REVIEW_FEEDBACK)
```

Comment:
There is a lot of stuff going on here. Can't you just do this:

```
actor_user = make_the_user
card = make_the_card

creators.log_card_moved_to_review_feedback(card, actor_user)
creators.log_card_moved_to_review_feedback(card, actor_user)

self.assertTrue(log_entries, 1)
```

Why should the creator care about any of that extra stuff? If we added a new state to cards (eg in_trusted_review)  then this test should still be totally relevent and correct. 

Always think about least privilage. With code things are always on a need to know basis. Every piece of code should know as little as possible about the surrounding system.  That's called loose coupling.